### PR TITLE
Printing of SCF

### PR DESCRIPTION
**Pull request by antoine-levitt**

### CODE REVIEW

File: src/scf/self_consistent_field.jl [hunk 1]
```diff
@@ -125,6 +127,8 @@ Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
     n_iter = 0
     energies = nothing
     ham = nothing
+    nonconsistent_energies = nothing
+    previous_energies = nothing
```
Comment (mfherbst) [inline src/scf/self_consistent_field.jl:131]: I think this should not live here, much rather the convergence / printing functors should cache that by themselves.

File: src/scf/self_consistent_field.jl [hunk 2]
```diff
@@ -170,10 +177,10 @@ Solve the Kohn-Sham equations with a SCF algorithm, starting at ρ.
         ρnext.real
     end
 
-    fpres = solver(fixpoint_map, ρout.real, maxiter; tol=min(10eps(T), tol / 10))
-    # Tolerance is only dummy here: Convergence is flagged by is_converged
+    # Tolerance and maxiter are only dummy here: Convergence is flagged by is_converged
     # inside the fixpoint_map. Also we do not use the return value of fpres but rather the
     # one that got updated by fixpoint_map
+    fpres = solver(fixpoint_map, ρout.real, typemax(Int); tol=0)
```
Comment (mfherbst) [inline src/scf/self_consistent_field.jl:183]: Sure tol=0 is a good idea. I tried that at some point and this caused numerical problems in *some* fixed point solvers, so I changed it to the value it has now. Have you tested with the crop and the custom anderson and the custom simple we have?

File: src/scf/self_consistent_field.jl [hunk 3]
```diff
@@ -29,14 +29,17 @@ function next_density(ham::Hamiltonian;
...
 
 function scf_default_callback(info)
-    E = info.energies === nothing ? Inf : sum(values(info.energies))
-    res = norm(info.ρout.fourier - info.ρin.fourier)
     if info.n_iter == 1
-        label = haskey(info.energies, "Entropy") ? "Free energy" : "Energy"
-        @printf "Iter   %-15s    ρout-ρin\n" label
-        @printf "----   %-15s    --------\n" "-"^length(label)
+        E_label = haskey(info.energies, "Entropy") ? "Free energy" : "Energy"
+        @printf "n     %-12s      ρout-ρin   Eₙ₋₁-Eₙ     diag\n" E_label
```
Comment (mfherbst) [inline src/scf/self_consistent_field.jl:34]: `diag` or `diag #it`? No strong opinion here.

Comment (mfherbst) [reply src/scf/self_consistent_field.jl:34]: Also diag is off by one space and for some reason I would indent the "Energy" label by one space (to not align with the - but the number) ... of course debatable. Also the capitalisation of the headers should be consistent (I'd just lower-case energy and entropy)

File: docs/src/guide/tutorial.jl [hunk 4]
```diff
@@ -45,10 +45,6 @@ basis = PlaneWaveBasis(model, Ecut; kgrid=kgrid)
 ## 3. Run the SCF procedure to obtain the ground state
 scfres = self_consistent_field(basis, tol=1e-8);
 
-# Note that DFTK by default applies the convergence tolerance `tol`
-# to the energy difference, so that the norm in the density difference
-# is not yet converged to 8 digits.
-#
```
Comment (mfherbst) [inline docs/src/guide/tutorial.jl:51]: Why do you want to remove that?

Comment (antoine-levitt) [reply docs/src/guide/tutorial.jl:51]: Because it's clearer now that there is the difference in energy why this stops

Comment (mfherbst) [reply docs/src/guide/tutorial.jl:51]: true. :+1:

File: src/scf/self_consistent_field.jl (hunk 1 above)
Comment (antoine-levitt) [reply src/scf/self_consistent_field.jl:131]: agree

File: src/scf/self_consistent_field.jl (hunk 2 above)
Comment (antoine-levitt) [reply src/scf/self_consistent_field.jl:183]: Well I'll see with the CI. We should just fix those solvers in that case

Comment (mfherbst) [reply src/scf/self_consistent_field.jl]: Seems good. I'll merge it.

**Pull request merged**